src/vm/wren_value.c

```diff
@@ -1071,17 +1063,43 @@ void wrenMarkObj(WrenVM* vm, Obj* obj)
 #endif
 }
 
-void wrenMarkValue(WrenVM* vm, Value value)
+void wrenExpandMarks(WrenVM* vm, GCState* state)
+{
+  for (Obj* obj = state->first; obj != NULL; obj = state->first)
+  {
+    state->first = obj->gcNext;
+
+    expandMark(vm, obj, state);
+  }
+}
+
+void wrenMarkObj(WrenVM* vm, Obj* obj, GCState* state)
+{
+  if (obj == NULL) return;
+
+  // Stop if the object is already marked so we don't get stuck in a cycle.
+  if (obj->marked) return;
```

Comment:
How about renaming this field `isBlack` to make the tri-color abstraction more explicit?